### ISSUE TITLE
Synergy scores in test data used to assess accuracy have information from entire dataset

### ISSUE THREAD

**Issue by creikey** [TechLabs-Aachen-e-V/WiSe20_Team_10_Main#1]
Because the test game data is used to calculate the synergy scores during the test game data evaluation, the random forest could be memorizing the relationship between the synergy scores and who wins then using that to overfit to test data, while being significantly less accurate in the real world

**Comment by fqhd** [comment]
> Because the test game data is used to calculate the synergy scores during the test game data evaluation, the random forest could be memorizing the relationship between the synergy scores and who wins then using that to overfit to test data, while being significantly less accurate in the real world

That's true, I can't believe they are using data from the testing set during training. This is very dissapointing :/

**Comment by creikey** [comment]
> > Because the test game data is used to calculate the synergy scores during the test game data evaluation, the random forest could be memorizing the relationship between the synergy scores and who wins then using that to overfit to test data, while being significantly less accurate in the real world
> 
> That's true, I can't believe they are using data from the testing set during training. This is very dissapointing :/

Yeah , I was unable to reproduce your results on professional league data. I used a different approach with a GRU and second by second game data and got to 82% accuracy on completely blind data without any processing of the validation set
